src/test/java/com/google/devtools/build/lib/bazel/repository/skylark/SkylarkRepositoryContextTest.java

```diff
@@ -131,7 +131,8 @@ protected void setUpContextForRule(
             ImmutableMap.of("FOO", "BAR"),
             downloader,
             1.0,
-            new HashMap<>());
+            new HashMap<>(),
```

Comment:
There should be tests for patch method with a stress on:
1) patch fragments applied only to the external repository and not outside.
2) what happens if patch file can not be found
3) a subpath inside external repository can not be found
4) patch error thrown from PatchUtil gets up the chain to the caller of repository_context.patch()

Reply:
Thanks for suggestion! I'll add those tests after I split this PR. ;)